File: finance_app/lib/data/models/category_breakdown.dart
class CategoryBreakdown {

  final String categoryId;
  final String categoryName;
  final double total;
  final double percentage;
  final int transactionCount;

  CategoryBreakdown({required this.categoryId,
  required this.categoryName,
  required this.total,
  required this.percentage,
  required this.transactionCount});

  factory CategoryBreakdown.fromJson(Map<String, dynamic> json){
    return CategoryBreakdown(
      categoryId: json['categoryId'],
      categoryName: json['categoryName'],
      total: json['total'].toDouble(),
      percentage: json['percentage'].toDouble(),
      transactionCount: json['transactionCount']
    );
  }

}

File: finance-tracker-backend/src/main/java/com/finance/tracker/transactions/domain/CategoryBreakdown.java
package com.finance.tracker.transactions.domain;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.util.UUID;

@Data
@AllArgsConstructor
public class CategoryBreakdown {

    private UUID categoryId;
    private String categoryName;
    private BigDecimal total;
    private BigDecimal percentage;
    private int transactionCount;

}
